llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -2391,6 +2688,8 @@ llvm::removeAllNonTerminatorAndEHPadInstructions(BasicBlock *BB) {
       ++NumDeadDbgInst;
     else
       ++NumDeadInst;
+    // RemoveDIs: erasing debug-info must be done manually.
```

Comment:
Do we need to `dropDbgValues` / `assert(has_no_dbg_values)` in the `if (Inst->isEHPad() || Inst->getType()->isTokenTy())` (outside the default context, above)?

Reply:
ping

Reply:
For EH pads this isn't an issue AFAIUI as getFirstInsertionPoint skips over those, and it would be an error for it to carry any variable assignments in either DbgValue or DPValue mode. I'm not sure about token types... I don't think I've ever seen one in the wild. I'd be inclined to say that we should ignore them, but there's no cost to _trying_ to drop them I guess, I've added a call on basis.